processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -73,7 +76,13 @@ public LookupReferencesManager(LookupConfig lookupConfig, final @Json ObjectMapp
   @LifecycleStart
   public void start()
   {
-    synchronized (lock) {
+    try {
```

Comment:
https://github.com/druid-io/druid/issues/3063 is a really nasty failure mode. With https://github.com/druid-io/druid/pull/3068 you can at least limit the effect to only lookups. I want a way to be able to interrupt misbehaving lookups.

The ReadWriteLock allows the start/stop to hold the Write part, and everything else to just hold the Read locks. This allows multiple management things to happen at the same time (as per the tests) such as some lookup being added while another is deleted.